### ISSUE TITLE
Support/documentation for actor initialization problem handling

### ISSUE THREAD

**Issue by PAStheLoD** [akka/alpakka-kafka#542]
Hello!

Using 0.22, and after browsing the issue tracker I think this is a request for enhancement. Currently it's either not possible (or at least "very" undocumented to figure out how) to catch initialization exceptions.

I'm thinking of these:
```
akka.actor.ActorInitializationException: akka://application/system/kafka-consumer-1: exception during creation
...
Caused by: org.apache.kafka.common.KafkaException: Failed to construct kafka consumer
...
Caused by: org.apache.kafka.common.config.ConfigException: No resolvable bootstrap urls given in bootstrap.servers
```

Even if I manually initialize the actor, since the exception happens in Akka, there's no way to handle it.

I see that the documentation [changed](https://github.com/akka/alpakka-kafka/commit/281a91eed81836975e9288c656d7d890b48f8cc1), and it used to recommend using a supervisor actor. The current recommendation of using watchTermination doesn't seem to work for termination due to initialization errors.

**Comment by 2m** [comment]
When creating a consumer stream with the wrong settings for the consumer actor:

```scala
val result: Future[Done] = Consumer
   .plainSource(..., ...)
   .runForeach(println)
```

The `result` Future will be failed with a `akka.kafka.ConsumerFailed` exception which you can handle and recover.

Granted, not every failure sets the cause of the `ConsumerFailed` exception. We should improve that.

**Comment by ennru** [comment]
Fixed with #617
